### PR TITLE
Add bridge indexer server

### PR DESCRIPTION
**Pull request by Bridgerz**
## Description 

This is the start of exposing a public api for the indexer data. 

There is currently only one route. 

## Test plan 

Tested locally.

### CODE REVIEW

File: crates/sui-bridge-indexer/src/server.rs [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+        per_page,
+        chain_id,
+        address,
+    } = params.0;
+
+    let offset = (page * per_page) as i64;
+    let limit = per_page as i64;
+
+    let sql = r#"
+        SELECT DISTINCT ON (tt.chain_id, tt.nonce)
```
Comment (longbowlu) [inline crates/sui-bridge-indexer/src/server.rs:87]: a quick comment: have we explored using diesel orm to construct the query?

File: crates/sui-bridge-indexer/src/server.rs [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+                WHEN 'Approved' THEN 2
+                WHEN 'Deposited' THEN 1
+                ELSE 0
+            END DESC
+        LIMIT $3 OFFSET $4;
+    "#;
+
+    let txn_sender_bytes = match address {
+        Some(ref addr) => {
+            let addr_trimmed = addr.trim_start_matches("0x");
```
Comment (longbowlu) [inline crates/sui-bridge-indexer/src/server.rs:119]: I'm not familiar with `trim_start_matches` and don't know its behavior in some cases, can we put this into a function and write some unit tests for it?

File: crates/sui-bridge-indexer/src/server.rs (hunk 1 above)
Comment (longbowlu) [inline crates/sui-bridge-indexer/src/server.rs:87]: what does this `DISTINCT ON` do along with
```
            CASE tt.status
                WHEN 'Claimed' THEN 3
                WHEN 'Approved' THEN 2
                WHEN 'Deposited' THEN 1
                ELSE 0
            END DESC
```
?

File: crates/sui-bridge-indexer/src/config.rs [hunk 3]
```diff
@@ -25,6 +25,8 @@ pub struct IndexerConfig {
     pub eth_sui_bridge_contract_address: String,
 
     pub metric_port: u16,
+
+    pub api_service_port: Option<u16>,
```
[the remark this answers is not in the document]
Comment (Bridgerz) [reply crates/sui-bridge-indexer/src/config.rs:29]: Leaning towards keeping this simple for now with the other custom indexer unification effort will probably change how this will work.

File: crates/sui-bridge-indexer/src/main.rs [hunk 4]
```diff
@@ -93,6 +94,15 @@ async fn main() -> Result<()> {
             tokio::time::Duration::from_secs(30),
         )),
     );
+
+    if config.api_service_port.is_some() {
+        let api_service_address = SocketAddr::new(
+            IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)),
+            config.api_service_port.unwrap(),
+        );
+        run_server(api_service_address, datastore.clone()).await?;
```
[the remark this answers is not in the document]
Comment (Bridgerz) [reply crates/sui-bridge-indexer/src/main.rs:103]: Have run locally and the server doesn't exit immediately. Awaiting the handle shouldn't cause it to be dropped right?

File: crates/sui-bridge-indexer/src/server.rs [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+struct QueryParams {
+    page: usize,
+    per_page: usize,
+    chain_id: i32,
+    address: Option<String>,
+}
+
+#[debug_handler]
+async fn get_token_transfers(
+    params: Query<QueryParams>,
```
[the remark this answers is not in the document]
Comment (Bridgerz) [reply crates/sui-bridge-indexer/src/server.rs:70]: // ?page=0&per_page=10&chain_id=1&address=0x1234

will add a comment

File: crates/sui-bridge-indexer/src/server.rs (hunk 2 above)
Comment (Bridgerz) [reply crates/sui-bridge-indexer/src/server.rs:119]: will do

File: crates/sui-bridge-indexer/src/server.rs (hunk 1 above)
Comment (Bridgerz) [reply crates/sui-bridge-indexer/src/server.rs:87]: The DISTINCT ON clause selects only one row for each unique chain_id and nonce combination that is determined by the ORDER BY clause. The ORDER BY clause prioritises rows by nonce in descending order and then ranks the tt.status values (Claimed, Approved, Deposited) so that the row with the most "recent" status is selected. 

TLDR: this ensures that for a given "token transfer", only one record is returned (instead of at most 3)
